### PR TITLE
free strategy - include failed hosts that were notified

### LINKED ISSUE #65254: The --force-handlers option if used with 'strategy: free' does not run handlers on all hosts.
##### SUMMARY

As of Ansible 2.7 the `--force-handlers` option is no longer running the handler/s on all hosts when used with strategy free. Only tested back to 2.5 and this was working as I would expect it to on 2.5 & 2.6.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

plugins/strategy 

##### ANSIBLE VERSION

```paste below
ansible 2.10.0.dev0
  config file = /Users/bdudas/dev-net/handlers/ansible.cfg
  configured module search path = [u'/Users/bdudas/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/bdudas/dev-net/handlers/ansible/lib/ansible
  executable location = /Users/bdudas/dev-net/handlers/ansible/bin/ansible
  python version = 2.7.16 (default, Oct 16 2019, 00:34:56) [GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.37.14)]
```

##### CONFIGURATION

```paste below
DEFAULT_STRATEGY(/Users/bdudas/dev-net/handlers/ansible.cfg) = free
HOST_KEY_CHECKING(/Users/bdudas/dev-net/handlers/ansible.cfg) = False
RETRY_FILES_ENABLED(/Users/bdudas/dev-net/handlers/ansible.cfg) = False
```

##### STEPS TO REPRODUCE

```yaml
---
- hosts: all
  gather_facts: false

  tasks:
    - command: hostname
      register: id
      notify: recovery_x

    - debug:
        var: id

    - name: failure
      fail: msg="Testing Failure"
      when: inventory_hostname == 'covfefe'

  handlers:
    - name: recovery_x
      shell: hostname
...
```
ansible-playbook <playbook.yml> --force-handlers

##### EXPECTED RESULTS

Handlers will run against all hosts when using the `--force-handlers` option. 

##### ACTUAL RESULTS

Handler is only run on 1 of the 2 target hosts.

```paste below

```

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
The free strategy was excluding failed hosts that were notified so the base class didn't run handlers on those hosts even when --force-handlers was used.

~WIP while I add a test for it to keep it from breaking in the future.~

Fixes #65254

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

**Comment by s-hertel** [comment]
This is largely a whitespace change because I added indentation to make the loop easier to read. The change without the indentation diff can be seen with https://github.com/ansible/ansible/pull/65576/files?w=1.

File: lib/ansible/plugins/strategy/free.py [hunk 1]
```diff
@@ -76,6 +80,7 @@ def run(self, iterator, play_context):
         list again, which would end up favoring hosts near the beginning of the
         list.
         '''
+        self.iterator = iterator
```
Comment (sivel) [inline lib/ansible/plugins/strategy/free.py]: I kind of question this line here.  I understand what you are doing and trying to achieve, but it feels kind of like a hack to inject this into class at this point...working around a shortcoming of the class.  This makes me feel like that should be something that is addressed.  Not sure, just offering opinions on my immediate reaction.

Comment (s-hertel) [reply lib/ansible/plugins/strategy/free.py]: @sivel Thanks for commenting. I'm not happy with this line either and it's why I attempted to use the TQM at first. I could instead pass the iterator to the filtering function. Do you think that would be better?

Comment (s-hertel) [reply lib/ansible/plugins/strategy/free.py]: My only concern with that is that it could break 3rd party strategy plugins.

**Pull request merged**